### PR TITLE
refactored everything

### PR DESCRIPTION
**Pull request by ShreyasRmsft**
refactored everything in preparation for adding offline support

### CODE REVIEW

File: Tasks/VsTestPlatformToolInstallerV1/task.loc.json [hunk 1]
```diff
@@ -13,35 +13,95 @@
...
       "visibleRule": "versionSelector = specificVersion"
+    },
+    {
+      "name": "customFeed",
+      "type": "string",
+      "label": "ms-resource:loc.input.label.customFeed",
+      "required": true,
+      "groupName": "packageSettings",
+      "helpMarkDown": "ms-resource:loc.input.help.customFeed",
+      "visibleRule": "packageFeedSelector = customFeed"
```
Comment (acesiddhu) [inline Tasks/VsTestPlatformToolInstallerV1/task.loc.json]: >visibleRule [](start = 7, length = 11)

hide the new inputs

File: Tasks/VsTestPlatformToolInstallerV1/nugetdownloadhelper.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        tl.debug(`Downloading Test Platform version ${testPlatformVersion} from ${packageSource} to ${downloadPath}.`);
+        this.consolidatedCiData.downloadStartTime = perf();
+        const resultCode = await nugetTool.exec();
+        this.consolidatedCiData.downloadEndTime = perf();
+
+        tl.debug(`Nuget.exe returned with result code ${resultCode}`);
+
+        if (resultCode !== 0) {
+            tl.error(tl.loc('NugetErrorCode', resultCode));
+            throw new Error(`Download failed with error code: ${resultCode}.`);
```
Comment (acesiddhu) [inline Tasks/VsTestPlatformToolInstallerV1/nugetdownloadhelper.ts]: >Download failed with error code [](start = 29, length = 31)

localized

Comment (acesiddhu) [inline Tasks/VsTestPlatformToolInstallerV1/nugetdownloadhelper.ts]: >Download failed with error code [](start = 29, length = 31)

localized

File: Tasks/VsTestPlatformToolInstallerV1/nugetfeedinstaller.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import * as perf from 'performance-now';
+import * as ci from './cieventlogger';
+import * as constants from './constants';
+import * as helpers from './helpers';
+import { NugetPackageVersionHelper } from './nugetpackageversionhelper';
+import { NugetDownloadHelper } from './nugetdownloadhelper';
+import { async } from 'q';
+
+export class NugetFeedInstaller {
+    public consolidatedCiData: { [key: string]: string; } = <{ [key: string]: string; }>{};
```
Comment (acesiddhu) [inline Tasks/VsTestPlatformToolInstallerV1/nugetfeedinstaller.ts]: >public [](start = 4, length = 6)

private

File: Tasks/VsTestPlatformToolInstallerV1/task.loc.json (hunk 1 above)
Comment (ShreyasRmsft) [reply Tasks/VsTestPlatformToolInstallerV1/task.loc.json]: done

**Pull request merged**